src/snowflake/snowpark/session.py

```diff
@@ -557,7 +557,13 @@ def __init__(
         conn: Union[ServerConnection, MockServerConnection, NopConnection],
         options: Optional[Dict[str, Any]] = None,
     ) -> None:
-        if len(_active_sessions) >= 1 and is_in_stored_procedure():
+        if (
+            len(_active_sessions) >= 1
+            and is_in_stored_procedure()
+            and not conn._get_client_side_session_parameter(
+                "ENABLE_CREATE_SESSION_IN_STORED_PROCS", False
```

Comment:
`ENABLE_CREATE_SESSION_IN_STORED_PROCS` is an internal param that will be only enabled for internal dbt runs.